### ISSUE TITLE
Form elements must have labels

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-next#1438]
Multiple component demos exhibit this same issue. Details related to this accessibility violation can be found here: https://dequeuniversity.com/rules/axe/3.0/label

Below are the URLs that illustrate the problem:

-  /components/Login/examples-full/
- /components/Switch/examples-full/
- /upgrade-examples/LoginUpgradeExamples/examples-full/

This is a `critical` a11y issue.

To learn more about how to build and test locally, check out: https://github.com/patternfly/patternfly-next#testing-for-accessibility

**As part of this issue, please complete the following items.** These will help reviewers know what issues were addressed in the PR, and also will help us keep our total a11y violation count down.

1. Run the a11y audit. Check if the violation returned is part of a PF3 component. For any violations in PF3 components, paste them in a comment below. If all violations are in the PF3 component, close this issue. Otherwise, continue with the next step...
1. Run the a11y audit, copy the issues from the audit that were fixed in the PR, and paste them into the PR description. 
    - Instructions for running the audit are here: https://github.com/patternfly/patternfly-next#testing-for-accessibility
    - Issues that are returned in the terminal look like this. Find the issues you are addressing and copy the following information provided for them or take a screencap to include in the PR.
      ```
      1 error found in: /upgrade-examples/AlertUpgradeExamples/examples-full/

      1) Buttons must have discernible text -- https://dequeuniversity.com/rules/axe/3.1/button-name -- impact: critical
      
      -------html violation instances-------

      <button class="btn btn-link dropdown-toggle" type="button" id="dropdownKebabRight" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
        <span class="fa fa-ellipsis-v"></span>
      </button> 
      ```
2. Reduce the threshold by number of issues being addressed 
    - This update is made [here](https://github.com/patternfly/patternfly-next/blob/master/test-a11y/config.js#L2)
    - The new value should be derived by: 
      - subtracting the number of issues you are addressing in the PR (e.g. the line that looks like this in the info you copied from the audit report in item 1 above):
        `5 errors found in: /components/Toolbar/examples-full/`
      - from current violations (e.g. this line in the audit report, near the end):
        `20 violations in total`

**Comment by jameslau** [comment]
Hey @jgiardino, should we star a new PR for this ticket in order to isolate the "upgrade examples"? Or should we just remove the upgrade portion from this ticket? @rachael-phillips, we had to close #1441 due to similar issues.

**Comment by jameslau** [comment]
@jgiardino, I went ahead and removed the "/upgrade-examples/LoginUpgradeExamples/examples-full/" task since we are no working on anything related to PF3 for ARIA.

**Comment by jameslau** [comment]
Due to the nature of the `aria-labelledby`, all labels for the switch component are driven dynamically by a set of handlebar rules, `aria-labelledby="switch-is-{{id}}"`. Because of this, the `aria-labelledby`can never match the elements ID because of the prefix "switch-is-" at the beginning. According to @mcoker, there is an enhancement on another ticketed issue that is being handled by @christiemolloy on https://github.com/patternfly/patternfly-next/pull/1473.

**Comment by jgiardino** [comment]
> @jgiardino, I went ahead and removed the "/upgrade-examples/LoginUpgradeExamples/examples-full/" task since we are no working on anything related to PF3 for ARIA.

That's fine as long as the issues were related to the PF3 component and not the PF4 component on that page. Since issues were also flagged for `/components/Login/examples-full/`, it's possible that the PF4 component on that page has issues. But it's also possible that if you fix the issues in `/components/Login/examples-full/`, that the PF4 issues in `/upgrade-examples/LoginUpgradeExamples/examples-full/` will automatically be fixed too. 

Regarding the switch, I'd say to focus on the issues flagged for `/components/Login/examples-full/` for now.

**Comment by jameslau** [comment]
I plugged '/upgrade-examples/LoginUpgradeExamples/examples-full/' back into the issue as a reference.
